notifier/microsoftteamsnotifier/notifier.go

```diff
@@ -0,0 +1,190 @@
+)
+
+const (
+	goFFLogo            = "https://raw.githubusercontent.com/thomaspoignant/go-feature-flag/main/logo_128.png"
+	microsoftteamsFooter         = "go-feature-flag"
+	colorDeleted        = "#FF0000"
+	colorUpdated        = "#FFA500"
+	colorAdded          = "#008000"
+	longMicrosoftTeamsAttachment = 35
+)
+
+type Notifier struct {
+	MicrosoftTeamsWebhookURL string
+
+	httpClient internal.HTTPClient
+	init       sync.Once
+}
+
+func (c *Notifier) Notify(diff notifier.DiffCache) error {
+	if c.MicrosoftTeamsWebhookURL == "" {
+		return fmt.Errorf("error: (Microsoft Teams Notifier) invalid notifier configuration, no " +
+			"MicrosoftTeamsWebhookURL provided for the microsoft teams notifier")
+	}
+
+	// init the notifier
+	c.init.Do(func() {
+		if c.httpClient == nil {
+			c.httpClient = internal.DefaultHTTPClient()
+		}
+	})
+
+	microsoftteamsURL, err := url.Parse(c.MicrosoftTeamsWebhookURL)
+	if err != nil {
+		return fmt.Errorf("error: (Microsoft Teams Notifier) invalid MicrosoftTeamsWebhookURL: %v", c.MicrosoftTeamsWebhookURL)
+	}
+
+	reqBody := convertToMicrosoftTeamsMessage(diff)
+	payload, err := json.Marshal(reqBody)
+	if err != nil {
+		return fmt.Errorf("error: (Microsoft Teams Notifier) impossible to read differences; %v", err)
+	}
+	request := http.Request{
+		Method: http.MethodPost,
+		URL:    microsoftteamsURL,
+		Body:   io.NopCloser(bytes.NewReader(payload)),
+		Header: map[string][]string{"Content-type": {"application/json"}},
+	}
+	response, err := c.httpClient.Do(&request)
+	if err != nil {
+		return fmt.Errorf("error: (Microsoft Teams Notifier) error: while calling webhook: %v", err)
+	}
+
+	defer func() { _ = response.Body.Close() }()
+	if response.StatusCode > 399 {
+		return fmt.Errorf("error: (Microsoft Teams Notifier) while calling microsoft teams webhook, statusCode = %d",
+			response.StatusCode)
+	}
+
+	return nil
+}
+
+func convertToMicrosoftTeamsMessage(diffCache notifier.DiffCache) microsoftteamsMessage {
+	hostname, _ := os.Hostname()
+	attachments := convertDeletedFlagsToMicrosoftTeamsMessage(diffCache)
+	attachments = append(attachments, convertUpdatedFlagsToMicrosoftTeamsMessage(diffCache)...)
+	attachments = append(attachments, convertAddedFlagsToMicrosoftTeamsMessage(diffCache)...)
+	currentDate := time.Now().Format("YYYY-MM-DD")
+	res := []map[string]interface{}{
+		"@type":    "MessageCard",
+		"@context": "https://schema.org/extensions",
+		"summary":  "Changes detected in your feature flag",
+		"sections": []map[string]interface{}{
+			{
+				"activityTitle":    "Changes detected in your feature flag",
+				"activitySubtitle": fmt.Sprintf("On flag file: *%s*", hostname),
+				"activityImage":    goFFLogo,
+				"text":             fmt.Sprintf("Changes detected in your feature flag file on: *%s*", hostname),
+				"facts": []map[string]string{
+					{"name": "Date", "value": currentDate},
+				},
+			},
+		},
+		"attachments": []Attachment{attachments}, 
```

Comment:
`Attachment` is not an existing type in your code, I guess you mean `attachment` here ?
```suggestion
		"attachments": []attachment{attachments}, 
```